pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -168,11 +306,425 @@ func (c *NPRecommendationController) syncNPRecommendation(key apimachinerytypes.
 		return err
 	}
 
-	klog.V(4).Infof("Syncing NP Recommendation %v", npReco)
-	// TODO (wshaoquan): add handling logic here
+	klog.V(4).Infof("Syncing NP Recommendation", "npReco", npReco)
+
+	switch npReco.Status.State {
+	case "", crdv1alpha1.NPRecommendationStateNew:
+		err = c.startJob(npReco)
+	case crdv1alpha1.NPRecommendationStateScheduled:
+		_, err = c.checkSparkApplicationStatus(npReco)
+	case crdv1alpha1.NPRecommendationStateRunning:
+		err = c.updateProgress(npReco)
+	case crdv1alpha1.NPRecommendationStateCompleted:
+		err = c.updateResult(npReco)
+	}
+	return err
+}
+
+func (c *NPRecommendationController) cleanupNPRecommendation(namespace string, sparkApplicationId string) error {
+	// Delete the Spark Application if exists
+	err := deleteSparkApplicationIfExists(c.kubeClient, namespace, sparkApplicationId)
+	if err != nil {
+		return fmt.Errorf("failed to delete the SparkApplication %s, error: %v", sparkApplicationId, err)
+	}
+	// Delete the RNP if exists
+	rnps, err := c.recommendedNPInformer.GetIndexer().ByIndex(idIndex, sparkApplicationId)
+	if err != nil {
+		return fmt.Errorf("failed to get the RecommendedNetworkPolicy when deleting, error: %v", err)
+	}
+	var aggregatedErrorMsg string
+	for _, obj := range rnps {
+		recommendedNetworkPolicy := obj.(*crdv1alpha1.RecommendedNetworkPolicy)
+		err = c.crdClient.CrdV1alpha1().RecommendedNetworkPolicies(namespace).Delete(context.TODO(), recommendedNetworkPolicy.Name, metav1.DeleteOptions{})
+		if err != nil {
+			aggregatedErrorMsg += fmt.Sprintf("failed to delete the RecommendedNetworkPolicy %s, error: %v\n", recommendedNetworkPolicy.Name, err)
```

Comment:
typically we don't want to have multi-line error messages. could collect failed name and err in 2 lists and call fmt when returning error.

Reply:
Make sense to me. Updated, thanks!